Kernel/FileSystem/InodeWatcher.cpp

```diff
@@ -62,37 +70,87 @@ KResultOr<size_t> InodeWatcher::read(FileDescription&, u64, UserOrKernelBuffer&
     return bytes_to_write;
 }
 
-KResultOr<size_t> InodeWatcher::write(FileDescription&, u64, const UserOrKernelBuffer&, size_t)
+KResult InodeWatcher::close()
 {
-    return EIO;
+    Locker locker(m_lock);
+
+    for (auto& entry : m_wd_to_watches) {
+        // NOTE: see comment in unregister_inode.
```

Comment:
What comment does this refer to? The only `unregister_inode` I see is in `TmpFS`.

Reply:
My bad, that's an outdated comment that's supposed to point to `unregister_by_inode`, and not important enough to reference.